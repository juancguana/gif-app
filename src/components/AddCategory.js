import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length >= 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input type='text' value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export { AddCategory };
