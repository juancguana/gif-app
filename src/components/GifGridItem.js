import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({ title, url }) => {
  return (
    <figure className='card animate__animated animate__fadeIn'>
      <img src={url} alt={title} />
      <figcaption>{title}</figcaption>
    </figure>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export { GifGridItem };
