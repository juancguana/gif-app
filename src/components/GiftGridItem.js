import React from 'react';
import PropTypes from 'prop-types';

const GiftGridItem = ({ title, url }) => {
  return (
    <figure className='card animate__animated animate__fadeIn'>
      <img src={url} alt={title} />
      <figcaption>{title}</figcaption>
    </figure>
  );
};

GiftGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GiftGridItem;
