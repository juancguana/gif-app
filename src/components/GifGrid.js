import React from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate__animated animate__fadeIn'>
        {category.toUpperCase()}
      </h3>
      {loading && (
        <figure className='animate__animated animate__flash loading'>
          <img src='/logo.png' height={100} alt='Logo' />
        </figure>
      )}
      <main className='card-grid'>
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </main>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export { GifGrid };
