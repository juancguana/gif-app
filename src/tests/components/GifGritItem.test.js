import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Test in <GifGridItem />', () => {
  const title = 'Un titulo';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should show the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a parragraph with the title', () => {
    const p = wrapper.find('figcaption');
    expect(p.text().trim()).toBe(title);
  });

  test('should have an image equal to url and alt by props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('should have class animate__fadeIn', () => {
    const figure = wrapper.find('figure');
    const className = figure.prop('className');
    expect(className.includes('animate__fadeIn')).toBe(true);
  });
});
