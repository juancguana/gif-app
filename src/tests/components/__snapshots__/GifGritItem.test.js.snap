// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem /> should show the component correctly 1`] = `
<figure
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <figcaption>
    Un titulo
  </figcaption>
</figure>
`;
